suite/tests/client-interface/avx512ctx.dll.c

```diff
@@ -55,53 +107,13 @@ bb_event(void *drcontext, void *tag, instrlist_t *bb, bool for_trace, bool trans
-                __asm__ __volatile__("vmovups %0, %%zmm6" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm7" : : "m"(buf));
-#ifdef X64
-                __asm__ __volatile__("vmovups %0, %%zmm8" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm9" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm10" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm11" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm12" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm13" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm14" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm15" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm16" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm17" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm18" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm19" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm20" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm21" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm22" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm23" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm24" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm25" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm26" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm27" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm28" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm29" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm30" : : "m"(buf));
-                __asm__ __volatile__("vmovups %0, %%zmm31" : : "m"(buf));
-#endif
-                dr_fprintf(STDERR, "Clobbering all mask registers\n");
-                __asm__ __volatile__("kmovw %0, %%k0" : : "m"(buf));
-                __asm__ __volatile__("kmovw %0, %%k1" : : "m"(buf));
-                __asm__ __volatile__("kmovw %0, %%k2" : : "m"(buf));
-                __asm__ __volatile__("kmovw %0, %%k3" : : "m"(buf));
-                __asm__ __volatile__("kmovw %0, %%k4" : : "m"(buf));
-                __asm__ __volatile__("kmovw %0, %%k5" : : "m"(buf));
-                __asm__ __volatile__("kmovw %0, %%k6" : : "m"(buf));
-                __asm__ __volatile__("kmovw %0, %%k7" : : "m"(buf));
+                opnd_get_reg(instr_get_dst(instr, 0)) == MARKER_REG) {
+                if (val1 == TEST1_MARKER) {
+                    clobber_avx512_state();
+                } else if (val1 == TEST2_MARKER) {
+                    dr_insert_clean_call(drcontext, bb, instr,
+                                         (void *)clobber_avx512_state, false, 0);
```

Comment:
OK, so we have a test of isolation of the app state: testing that the client won't clobber it.  I think we also need a test that the client can properly access the app state.

The other type of test is state preservation optimization and inlining: do the existing tests need to be augmented in any way for 512?

Reply:
I think we're covered well now with the latest update. But - I do want to review existing ones, for example I discovered that we are not really testing AVX state and signal context (linux.sigcontext). Specifically for clean calls, one test already had support for zmm, an outside contributor had added that a while ago, but the test is limited. Then client.cleancall-opt-1 should get support both for AVX and AVX-512. Both is missing there. I have added a comment. So opt/inline is not well tested, though it does test xmm registers, but it doesn't run on the AVX-512 part of context preservation. Looking at this now.

Reply:
I've also added an AVX-512 derivative of the client.cleancall-opt-1 test. The test is a separate binary so we have tests for both versions of the state preservation code paths.